csv/src/main/java/org/apache/metamodel/csv/CsvWriter.java

```diff
@@ -72,23 +72,38 @@ public String buildLine(String[] nextLine) {
     }
 
     private boolean stringContainsSpecialCharacters(String line) {
-        return line.indexOf(_configuration.getQuoteChar()) != -1 || line.indexOf(_configuration.getEscapeChar()) != -1;
+        boolean result = line.indexOf(_configuration.getQuoteChar()) != -1
+                || line.indexOf(_configuration.getEscapeChar()) != -1;
+        if (!result) {
+            result = _configuration.getQuoteChar() == CsvConfiguration.NOT_A_CHAR
+                    && _configuration.getEscapeChar() != CsvConfiguration.NOT_A_CHAR;
```

Comment:
You're right something looks funky here. I think it should be ```... && line.indexOf(quoteChar) != -1``` ... let me check with the unittest etc.